flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/UnionInputGate.java

```diff
@@ -189,11 +189,11 @@ public void requestPartitions() throws IOException, InterruptedException {
 
 		bufferOrEvent.setChannelIndex(channelIndexOffset + bufferOrEvent.getChannelIndex());
 
-		return Optional.ofNullable(bufferOrEvent);
+		return Optional.of(bufferOrEvent);
 	}
 
 	@Override
-	public Optional<BufferOrEvent> pollNextBufferOrEvent() throws IOException, InterruptedException {
+	public Optional<BufferOrEvent> pollNextBufferOrEvent() throws UnsupportedOperationException {
```

Comment:
Do you need to declare to throw this exception? Do we want to declare `RuntimeExceptions`?

Reply:
we don't need to - I thought this makes it more explicit if anyone tries to use this method in an IDE without looking at the code (which is a mistake imho, but commonly done)

Reply:
as you prefer